extensions/amp-animation/0.1/amp-animation.js

```diff
@@ -306,14 +333,29 @@ export class AmpAnimation extends AMP.BaseElement {
       return null;
     }
 
-    return this.createRunner_(opt_args).then(runner => {
-      this.runner_ = runner;
-      this.runner_.onPlayStateChanged(this.playStateChanged_.bind(this));
-      this.setupScrollboundAnimations_();
+    return this.maybeCreateRunner_(opt_args).then(() => {
       this.runner_.start();
     });
   }
 
+  /**
+   * Creates the runner but animations will not start.
+   * @param {?JsonObject=} opt_args
+   * @return {!Promise}
+   * @private
+   */
+  maybeCreateRunner_(opt_args) {
+    if (!this.runnerPromise_ || !this.runner_) {
```

Comment:
Not sure how to read this: why would one exist but the other would not? Further, if `runner_` exists but promise doesn't, why do we recreate the runner? Shouldn't the previous runner be destroyed/stopped/etc? It will continue playing otherwise. This is from reading the code, so maybe just some asserts missing?